### DISCUSSION TITLE
Camel Case Json Serialize / Deserialize

### DISCUSSION THREAD

**Discussion by CodingFlow** [nats-io/nats.net#700] [Q&A]
Having the classic issue of needed to set camel case json serialization/deserialization for PascalCase C# classes. I can see the [setting](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/customize-properties#use-a-built-in-naming-policy) for it in System.Text.Json, but I don't how to set it for nats.net. I looked over the serialization section of the nets.net documentation, but got quite lost trying to figure it out. I'm trying to set it up for `ConsumeAsync`. Thanks in advance for any help!

**Comment by mtmk** [comment]
have a look at the examples here if you are using generated serialization https://github.com/nats-io/nats.net/blob/main/src/NATS.Client.JetStream/NatsJSJsonSerializer.cs#L98

if you need help setting up a custom serializer in NATS .NET if you can drop s minimal Program.cs here I can try to help.
